### DISCUSSION TITLE
Using the Java agent method resulted in the metadata being updated and the tarcepartment being generated during the GRPC callback

### DISCUSSION THREAD

**Discussion by WuChuhan** [open-telemetry/opentelemetry-java-instrumentation#10913] [Q&A]
I used a Java agent to start OpenTelemetry in the project, but initiating another GRPC request during the GRPC callback process would cause a change in the traceid. Is it because every time we arrive at a new GRPC client, a new metadata is added, and during the last GRPC callback process, the traceparent information cannot be passed down, and the agent method itself cannot cover this scenario?

**Comment by laurit** [comment]
Please provide a minimal sample app along with any necessary instructions that allows for reproducing this issue.

**Reply by WuChuhan** [reply]
`public String callSayHello() {
        logger.info("==============callSayHello start==============");
        doSomething();
        HelloRequest request = HelloRequest.newBuilder().setGreeting("world").build();
        HelloResponse response;
        try {
            response = blockingStub.sayHello(request);
            logger.info("callSayHello get reply: " + response.getReply());
        } catch (StatusRuntimeException e) {
            logger.warning("RPC failed: " +  e.getStatus());
            return "fail";
        }
        return "success";
    }

@GetMapping("/callSayHelloClientStream")
    public String callSayHelloClientStream() throws InterruptedException {
        logger.info("==============callSayHelloClientStream start==============");
        final CountDownLatch finishLatch = new CountDownLatch(1);

        StreamObserver<HelloResponse> responseObserver = new StreamObserver<HelloResponse>() {
            @Override
            public void onNext(HelloResponse helloResponse) {
                HelloRequest request = HelloRequest.newBuilder().setGreeting("world2").build();
                HelloResponse response = blockingStub.sayHello(request);
                logger.info("CallSayHelloClientStream get reply: " + helloResponse.getReply());
            }

            @Override
            public void onError(Throwable t) {
                logger.warning("CallSayHelloClientStream Failed: " + Status.fromThrowable(t));
                finishLatch.countDown();
            }

            @Override
            public void onCompleted() {
                logger.info("Finished CallSayHelloClientStream");
                finishLatch.countDown();
            }
        };

        StreamObserver<HelloRequest> requestObserver = asyncStub.sayHelloClientStream(responseObserver);
        try {
            for (int i = 0; i < 5; i++) {
                HelloRequest request = HelloRequest.newBuilder().setGreeting("world" + i).build();
                requestObserver.onNext(request);
                if (finishLatch.getCount() == 0){
                    return "success";
                }
            }
        } catch (StatusRuntimeException e) {
            requestObserver.onError(e);
            return "fail";
        }

        // Mark the end of requests
        requestObserver.onCompleted();

        if (!finishLatch.await(1, TimeUnit.MINUTES)) {
            logger.warning("CallSayHelloClientStream can not finish within 1 minutes");
        }

        return "success";
    }`

**Reply by WuChuhan** [reply]
When I make a request again in the GRPC callback, there is no traceparent in the metadata, which makes it impossible to achieve traceability.

**Reply by laurit** [reply]
Please provide a fully functional sample app as a github repository or a zip along with any instructions necessary to reproduce your issue.
